pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,75 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/machine/libmachine/drivers"
+)
+
+var (
+	pullSecret = `apiVersion: v1
+data:
+  .dockerconfigjson: %s
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: openshift-config
+type: kubernetes.io/dockerconfigjson`
+
+	// This command timeout in 80 secs if not able to replace the pull
+	// secret. We first start the kubelet and wait till the api server is
+	// up and running by hitting it with replace command.
+	replacePullSecretCmd = `timeout 80 bash -c 'sudo systemctl start kubelet && \
+until oc --config /tmp/kubeconfig replace -f /tmp/pull-secret.yaml 2>/dev/null 1>&2; \
+do echo "Waiting for recovery apiserver to come up."; sleep 1; done && \
+sudo systemctl stop kubelet && \
+sudo crictl stopp $(sudo crictl pods -q) && \
+sudo crictl rmp $(sudo crictl pods -q)'`
```

Comment:
I understand you want to keep the pull secret update self-contained, however starting the kubelet, and then stopping it and removing the misc pods seems quite heavyweight. Since the only time we are going to update the pull secret is going to be during the `start` cmd, can we let `machine.start` start the kubelet process, and only call `oc replace`? Or do we need to restart the kubelet anyway because it can't deal with changes to the pull secret dynamically?

Reply:
@cfergeau didn't get you, to replace the pull secret we need to have an apiserver which is why we are starting the kubelet and as soon as we have that then we replace it and again stop it.

Reply:
What is happening is that in cmd.Start(), we start kubelet, replace the pull secret, stop kubelet, cleanup pods, and then start again kubelet (later in Start()). I was wondering if the restart is mandatory for the changes to take effect, or if we could optimize this to delay start up as little as possible?

Reply:
@cfergeau after replace the pull secret restart is mandatory as per conversion I had on aos-devel ML but we can try to start the kubelet replace it and keep it running and see if that also work.

Reply:
oh, if you were told that this is mandatory, I'm fine with it, I'd document it either in a comment or in the commit log.